json.go

```diff
@@ -0,0 +1,100 @@
+package clover
+
+import (
+	"os"
+	"encoding/json"
+	"bufio"
+	"io/ioutil"
+	"errors"
+)
+
+// Exporting errors
+var (
+	ErrImportPathUnreachable = errors.New("the import path can't be reached")
+	ErrExportPathExists = errors.New("the export path already exists")
+	ErrExportCollectionEmpty = errors.New("the collection is empty")
+
+)
+
+func pathExists(path string) bool{
+	_, err := os.Stat(path)
+	return err == nil
+}
+
+// Exports a `collection` to JSON format in `filePath`
+func (db *DB) ExportCollection(collectionName string, exportPath string) error{
+	exists, err := db.HasCollection(collectionName)
+	if err != nil{
+		return err
+	}
+	if !exists{
+		return ErrCollectionNotExist
+	}
+	if pathExists(exportPath) {
+    	return ErrExportPathExists
+	}
+
+	count, err := db.Query(collectionName).Count()
```

Comment:
I think we could get rid of the additional call to `Count()`. Since anyway you have to call `FindAll()`, you could use len(result) to assert if the collection is non-empty. The overhead will be the same, since in the case you have an empty collection, no work will be done in the `FindAll()`, while if you have some results, you will always use them.

Reply:
Hello @ostafen, just changed this as proposed! The idea behind using `Count()` was to stick to the library functions. For future implementations, a more cost-effective `Count()` solution could be a good idea as well as this may be a common use case for this function